core/lib/slider.ts

```diff
@@ -0,0 +1,569 @@
+import {computed, writable} from '@amadeus-it-group/tansu';
+import type {WidgetsCommonPropsAndState} from './commonProps';
+import type {ConfigValidator, PropsConfig} from './services';
+import {bindableDerived, registrationArray, writablesForProps} from './services';
+import type {ReadableSignals} from './services/stores';
+import {mergeConfigStores, normalizeConfigStores, stateStores} from './services/stores';
+import {typeArray, typeBoolean, typeFunction, typeNumber, typeString} from './services/writables';
+import type {Directive, Widget} from './types';
+import {noop} from './utils';
+
+export interface SliderCommonPropsAndState extends WidgetsCommonPropsAndState {
+	/**
+	 * Minimum value that can be assigned to the slider
+	 */
+	min: number;
+
+	/**
+	 * Maximum value that can be assigned to the slider
+	 */
+	max: number;
+
+	/**
+	 * Unit value between slider steps
+	 */
+	stepSize: number;
+
+	/**
+	 * If `true` slider value cannot be changed but the slider is still focusable
+	 */
+	readonly: boolean;
+
+	/**
+	 * If `true` slider value cannot be changed and the slider cannot be focused
+	 */
+	disabled: boolean;
+
+	/**
+	 * If `true` is vertically positioned otherwise it is horizontal
+	 */
+	vertical: boolean;
+
+	/**
+	 * Dom Rectangle of the slider
+	 */
+	sliderDomRect: DOMRect;
+
+	/**
+	 * Dom Rectangle of the minimum label
+	 */
+	minLabelDomRect: DOMRect;
+
+	/**
+	 * Dom Rectangle of the maximum label
+	 */
+	maxLabelDomRect: DOMRect;
+
+	/**
+	 * Current slider values
+	 */
+	values: number[];
+}
+
+export interface SliderState extends SliderCommonPropsAndState {
+	/**
+	 * Progress bar left position in %
+	 */
+	progressLeft: number[];
+
+	/**
+	 * Progress bar bottom position in %
+	 */
+	progressBottom: number[];
+
+	/**
+	 * Progress bar height in %
+	 */
+	progressHeight: number[];
+
+	/**
+	 * Progress bar width in %
+	 */
+	progressWidth: number[];
+
+	/**
+	 * Sorted slider values
+	 */
+	sortedValues: number[];
+
+	/**
+	 * Mix label left offset in %
+	 */
+	mixLabelLeft: number;
```

Comment:
I don't understand neither naming nor documentation for these. What does it do?

Reply:
will add the 'position' to the variable name

Reply:
nah, either you should add `offset` or `position` to all props or to none. My comment was mostyl about `mixLabel` part.